projects/js-packages/connection/components/disconnect-dialog/index.jsx

```diff
@@ -45,54 +69,211 @@ const DisconnectDialog = props => {
 	}, [ apiRoot, apiNonce ] );
 
 	/**
-	 * Open the Disconnect Dialog.
+	 * Initialize tracks with user data.
+	 * Should run when we have a connected user.
 	 */
-	const openModal = useCallback(
-		e => {
-			e && e.preventDefault();
-			setOpen( true );
-		},
-		[ setOpen ]
-	);
+	useEffect( () => {
+		if ( connectedUser && connectedUser.ID && connectedUser.login ) {
+			jetpackAnalytics.initialize( connectedUser.ID, connectedUser.login );
+		}
+	}, [ connectedUser, connectedUser.ID, connectedUser.login ] );
 
 	/**
-	 * Close the Disconnect Dialog.
+	 * Run when the disconnect dialog is opened
 	 */
-	const closeModal = useCallback(
-		e => {
-			e && e.preventDefault();
-			setOpen( false );
+	useEffect( () => {
+		if ( isOpen ) {
+			jetpackAnalytics.tracks.recordEvent( 'jetpack_disconnect_dialog_open', defaultTracksArgs );
+		}
+	}, [ isOpen, defaultTracksArgs ] );
+
+	/**
+	 * Keep track of the steps that are presented
+	 */
+	useEffect( () => {
+		if ( isDisconnected && ! isProvidingFeedback && ! isFeedbackProvided ) {
```

Comment:
From the perspective of using these events in funnels, etc, it might make more sense to use a single event and vary a specific event property based on the current step.
This isn't a super strong opinion on my part though, so your call.

Reply:
Sounds good to me! I'll update this to use the same event name and vary a `step` property. Thanks for the recommendation here.